File: Array/121-Best-Time-to-Buy-and-Sell-Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        buy=sell=prices[0]
        res=0
        for price in prices:
            if price<buy:
                buy =price
                sell=price
            else:
                sell=price
            res=max(res,sell-buy)
        return res

File: Random/Arrays/121-Best-Time-to-Buy-and-Sell-Stock.ts
function maxProfit(prices: number[]): number {
    const len=prices.length;
    let buy=prices[0];
    let profit=0,max_profit=0;

    for(let i=1;i<len;i++){
        if(prices[i]<=buy)buy=prices[i];
        else{
            profit=prices[i]-buy;
            max_profit=Math.max(profit,max_profit);
        }
    }
    return max_profit;
};

